### PR TITLE
Feature/credit service

### PR DESCRIPTION
**Pull request by harsha-wedaa**
Credit service sync with wedaa-Server.

### CODE REVIEW

File: src/utils/creditcore.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+async function addTransactioLog(data, accessToken) {
```
Comment (rohith-wedaa) [inline src/utils/creditcore.js]: **addTransactioLog** typo - **n** missing

File: src/designer/core.js [hunk 2]
```diff
@@ -30,12 +31,47 @@ exports.prototype = async function (blueprintInfo) {
...
+            "blueprintId":blueprintInfo.blueprintId
+        }
+        var credits = {
+            "userId":userId,
+            "creditsAvailable": -1*aiservicesCount,
+            "creditsUsed": aiservicesCount
+        }
+        // TODO: Error Handling must be taking care!
+        await creditCore.addTransactioLog(transaction,accessToken)
+        await creditCore.updateUserCredit(credits,accessToken)
```
Comment (rohith-wedaa) [inline src/designer/core.js]: put this block in a service method, with required arguments.

**Review by rohith-wedaa — CHANGES_REQUESTED** [review]

File: src/designer/core.js [hunk 3]
```diff
@@ -30,12 +31,47 @@ exports.prototype = async function (blueprintInfo) {
...
+                if (
+                    service?.data &&
+                    service?.data?.dbmlData &&
+                    service?.data?.description
+                ) {
+                    aiservicesCount++;
+                }
+            }
+        });
+    }
```
Comment (raxkumar-wedaa) [inline src/designer/core.js]: Updated this to 
const dbmlCount = Object.values(services).filter(service => service.dbmlData).length;

File: src/designer/core.js [hunk 4]
```diff
@@ -30,12 +31,47 @@ exports.prototype = async function (blueprintInfo) {
...
+        // TODO: Error Handling must be taking care!
+        await creditCore.addTransactioLog(transaction,accessToken)
+        await creditCore.updateUserCredit(credits,accessToken)
+        }
     // Form a context object which will contain the information to be passed to generate zip & update code_generation
     var context = {
         userId: userId,
         codeGenerationId: codeGenerationId,
+        AIServices:aiservicesCount,
+        accessToken:accessToken
```
Comment (raxkumar-wedaa) [inline src/designer/core.js]: Removing this from here and adding a new creditContext

File: src/services/blueprintService.js [hunk 5]
```diff
@@ -279,3 +279,28 @@ exports.getBlueprintById = function (blueprintId) {
             throw error;
         });
 };
+
+exports.getData = async function (req, res) {
```
Comment (raxkumar-wedaa) [inline src/services/blueprintService.js]: Updated the method name to getMetadata

**Review by rohith-wedaa — APPROVED** [review]

**Pull request merged**